### PR TITLE
Refactoring of docker-runc

### PR DESCRIPTION
**Pull request by grisu48**
Refactoring on runc:

* Move the library function for runc to it's test module
* Move check for bsc#1152508 to the docker engine test
* Use tumbleweed instead of alpine
* Streamline test run

- Related ticket: https://progress.opensuse.org/issues/159021
- Verification runs: [Tumbleweed](https://duck-norris.qe.suse.de/tests/14614) | [15-SP5](https://duck-norris.qe.suse.de/tests/14608) | [15-SP4](https://duck-norris.qe.suse.de/tests/14609) | [15-SP3](https://duck-norris.qe.suse.de/tests/14610) | [15-SP2](https://duck-norris.qe.suse.de/tests/14611) | [12-SP5](https://duck-norris.qe.suse.de/tests/14612)

### CODE REVIEW

File: tests/containers/docker_runc.pm [hunk 1]
```diff
@@ -24,43 +25,82 @@ sub run {
...
+            select_console 'root-console';
+            trup_call("pkg install $runc");
+            check_reboot_changes;
+        } else {
+            zypper_call("in $runc");
+        }
+    }
+    record_info("$runc", script_output("$runc -v"));
+    # Create root filesystem for the test container. We need docker for this preparation step.
     assert_script_run('mkdir rootfs');
```
Comment (ricardobranco777) [inline tests/containers/docker_runc.pm:43]: We're no longer removing this directory in `cleanup()`;

File: tests/containers/docker_runc.pm [hunk 2]
```diff
@@ -24,43 +25,82 @@ sub run {
...
+    record_info 'Test #1', 'Installation and test preparation';
+    if (script_run("which $runc") != 0) {
+        if (is_transactional) {
+            select_console 'root-console';
+            trup_call("pkg install $runc");
+            check_reboot_changes;
+        } else {
+            zypper_call("in $runc");
+        }
+    }
```
Comment (ricardobranco777) [inline tests/containers/docker_runc.pm:40]: You can now `use containers::common "install_packages"` like this:
```suggestion
    install_packages($runc);
```

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/common.pm#L353

**Review by ricardobranco777 — APPROVED** [review]
A couple of suggestions.

A VR for Tumbleweed would be nice to make sure we have no issue in upstream.

File: tests/containers/docker_runc.pm (hunk 2 above)
Comment (grisu48) [reply tests/containers/docker_runc.pm:40]: oh nice, that simplifies this whole block!

File: tests/containers/docker_runc.pm (hunk 1 above)
Comment (grisu48) [reply tests/containers/docker_runc.pm:43]: Good catch.

Comment (grisu48) [reply tests/containers/docker_runc.pm:43]: I'm also adding a `rm -rf rootfs` here, so the test would not fail in case the directory already exists.

**Comment by grisu48** [comment]
Marking as draft until VRs are passing. Rescheduled all VRs and added Tumbleweed as well.

**Comment by grisu48** [comment]
I squashed the three commits into one.

**Comment by grisu48** [comment]
New VR passed: https://duck-norris.qe.suse.de/tests/14614

**Pull request merged**